Client/Frontend/Browser/FindInPageBar.swift

```diff
@@ -0,0 +1,136 @@
+    private let searchText = UITextField()
+    private let matchCountView = UILabel()
+    private let previousButton = UIButton()
+    private let nextButton = UIButton()
+
+    var currentResult = 0 {
+        didSet {
+            matchCountView.text = "\(currentResult)/\(totalResults)"
+        }
+    }
+
+    var totalResults = 0 {
+        didSet {
+            matchCountView.text = "\(currentResult)/\(totalResults)"
+            previousButton.enabled = totalResults > 1
+            nextButton.enabled = previousButton.enabled
+        }
+    }
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        backgroundColor = UIColor.whiteColor()
+
+        searchText.addTarget(self, action: "didTextChange:", forControlEvents: UIControlEvents.EditingChanged)
+        searchText.textColor = UIColor(rgb: 0xe66000)
+        searchText.font = UIConstants.DefaultChromeFont
+        searchText.autocapitalizationType = UITextAutocapitalizationType.None
+        searchText.autocorrectionType = UITextAutocorrectionType.No
+        addSubview(searchText)
+
+        matchCountView.textColor = UIColor.lightGrayColor()
+        matchCountView.font = UIConstants.DefaultChromeFont
+        matchCountView.hidden = true
+        addSubview(matchCountView)
+
+        previousButton.setImage(UIImage(named: "find_previous"), forState: UIControlState.Normal)
+        previousButton.setTitleColor(UIColor.blackColor(), forState: UIControlState.Normal)
+        previousButton.accessibilityLabel = NSLocalizedString("Previous in-page result", tableName: "FindInPage", comment: "Accessibility label for previous result button in Find in Page Toolbar.")
+        previousButton.addTarget(self, action: "didFindPrevious:", forControlEvents: UIControlEvents.TouchUpInside)
+        addSubview(previousButton)
+
+        nextButton.setImage(UIImage(named: "find_next"), forState: UIControlState.Normal)
+        nextButton.setTitleColor(UIColor.blackColor(), forState: UIControlState.Normal)
+        nextButton.accessibilityLabel = NSLocalizedString("Next in-page result", tableName: "FindInPage", comment: "Accessibility label for next result button in Find in Page Toolbar.")
+        nextButton.addTarget(self, action: "didFindNext:", forControlEvents: UIControlEvents.TouchUpInside)
+        addSubview(nextButton)
+
+        let closeButton = UIButton()
+        closeButton.setImage(UIImage(named: "find_close"), forState: UIControlState.Normal)
+        closeButton.setTitleColor(UIColor.blackColor(), forState: UIControlState.Normal)
+        closeButton.accessibilityLabel = NSLocalizedString("Done", tableName: "FindInPage", comment: "Done button in Find in Page Toolbar.")
+        closeButton.addTarget(self, action: "didPressClose:", forControlEvents: UIControlEvents.TouchUpInside)
+        addSubview(closeButton)
+
+        let topBorder = UIView()
+        topBorder.backgroundColor = UIColor(rgb: 0xEEEEEE)
+        addSubview(topBorder)
```

Comment:
Nit: Move the styling attributes into a UX struct since it seems to be the convention so far.